addons/stock/models/stock_picking.py

```diff
@@ -519,6 +519,10 @@ def write(self, vals):
                         pickings_to_not_autoconfirm |= picking
                         break
             (self - pickings_to_not_autoconfirm)._autoconfirm_picking()
+        if vals.get('picking_type_id'):
```

Comment:
It shouldn't be in the onchange_picking_type ? it will be more suitable for the user, no ?